src/silo/file_reader/sam_reader.test.cpp

```diff
@@ -0,0 +1,109 @@
+
+#include "silo/config/preprocessing_config.h"
+#include "silo/sequence_file_reader/sam_format_exception.h"
+#include "silo/sequence_file_reader/sam_reader.h"
+
+using silo::sequence_file_reader::SamFormatException;
+using silo::sequence_file_reader::SamReader;
+
+TEST(SamReader, shouldReadSamFile) {
+   const std::string file_content{
+      "1.1\t99\tNC_045512.2\t10\t60\t5S246M\t="
+      "\t201\t404\tTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCT\tCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC5CCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC5CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCC\tNM:i:1\tMD:Z:197C48\tMC:Z:247M\tAS:i:241\tXS:i:0\n"
+      "1.3\t99\tNC_045512.2\t5\t60\t47S204M\t="
+      "\t209\t400\tTATACCTTCCCAGGTAACAAACCAACCAACTTTTTTTTT\tCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCC5CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC5CCCCCCCCCC"
+      "CCCCCCC*"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCC\tNM:i:1\tMD:Z:196C7\tMC:Z:236M\tAS:i:199\tXS:i:0\tSA:Z:NC_045512.2,"
+      "7769,+,30M221S,60,0;"
+   };
+
+   SamReader under_test(file_content);
+
+   auto entry = under_test.nextEntry();
+   EXPECT_TRUE(entry.has_value());
+   auto [key, offset, genome] = entry.value();
+   EXPECT_EQ(key, "1.1");
+   EXPECT_EQ(offset, 10);
+   EXPECT_EQ(genome, "TATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCT");
+
+   entry = under_test.nextEntry();
+   EXPECT_TRUE(entry.has_value());
+   key = entry.value().key;
+   offset = entry.value().offset;
+   genome = entry.value().sequence;
+   EXPECT_EQ(key, "1.3");
+   EXPECT_EQ(offset, 5);
+   EXPECT_EQ(genome, "TATACCTTCCCAGGTAACAAACCAACCAACTTTTTTTTT");
+
+   entry = under_test.nextEntry();
+   EXPECT_FALSE(entry.has_value());
+}
+
+TEST(SamReader, shouldReadSamFileWithBlankLines) {
+   const std::string file_content{
+      "1.1\t99\tNC_045512.2\t10\t60\t5S246M\t="
+      "\t201\t404\tTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCT\tCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC5CCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC5CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCC\tNM:i:1\tMD:Z:197C48\tMC:Z:247M\tAS:i:241\tXS:i:0\n"
+      "\n\n1.3\t99\tNC_045512.2\t5\t60\t47S204M\t="
+      "\t209\t400\tTATACCTTCCCAGGTAACAAACCAACCAACTTTTTTTTT\tCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCC5CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC5CCCCCCCCCC"
+      "CCCCCCC*"
+      "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+      "CCCCCCCCCCCCCCCCCCCC\tNM:i:1\tMD:Z:196C7\tMC:Z:236M\tAS:i:199\tXS:i:0\tSA:Z:NC_045512.2,"
+      "7769,+,30M221S,60,0;"
+   };
```

Comment:
Let's also make this a raw string? :)

Reply:
I anticipated this comment xD but I'm not sure it would help because the tabs would need to be actual tabs and it would take even more space, if I understood how raw strings work correctly?

Reply:
You can type actual tabs into a raw string. At least it would be readable then ;)